BUG: pytest scipy/linalg/tests/test_extending.py fails with Cython file compilation error

### Describe your issue.

`pytest scipy/linalg/tests/test_extending.py` fails with the attached error. I am using editable install as per the contributing documentation.


 Out of the 65k+ test cases this is the only test that fails. 

### Reproducing Code Example

```python
pytest scipy/linalg/tests/test_extending.py
```


### Error message

```shell
==================================================================================================== test session starts =====================================================================================================
platform darwin -- Python 3.11.8, pytest-8.1.1, pluggy-1.4.0
rootdir: /Users/tuhinsharma/Documents/Git/scipy
configfile: pytest.ini
plugins: cov-5.0.0, hypothesis-6.100.1, anyio-4.3.0, xdist-3.6.0, timeout-2.3.1
collected 1 item

scipy/linalg/tests/test_extending.py F                                                                                                                                                                                 [100%]

========================================================================================================== FAILURES ==========================================================================================================
________________________________________________________________________________________________________ test_cython _________________________________________________________________________________________________________

tmp_path = PosixPath('/private/var/folders/24/q83vy1q54_181scf16svmls40000gn/T/pytest-of-tuhinsharma/pytest-10/test_cython0')

    @pytest.mark.skipif(platform.machine() in ["wasm32", "wasm64"],
                        reason="Can't start subprocess")
    @pytest.mark.skipif(cython is None, reason="requires cython")
    def test_cython(tmp_path):
        srcdir = os.path.dirname(os.path.dirname(__file__))
>       extensions, extensions_cpp = _test_cython_extension(tmp_path, srcdir)

srcdir     = '/Users/tuhinsharma/Documents/Git/scipy/scipy/linalg'
tmp_path   = PosixPath('/private/var/folders/24/q83vy1q54_181scf16svmls40000gn/T/pytest-of-tuhinsharma/pytest-10/test_cython0')

scipy/linalg/tests/test_extending.py:17:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
scipy/_lib/_testutils.py:295: in _test_cython_extension
    subprocess.check_call(["meson", "compile", "-vv"], cwd=target_dir)
        build_dir  = PosixPath('/private/var/folders/24/q83vy1q54_181scf16svmls40000gn/T/pytest-of-tuhinsharma/pytest-10/test_cython0/linalg/tests/_cython_examples')
        mod_name   = 'linalg'
        srcdir     = '/Users/tuhinsharma/Documents/Git/scipy/scipy/linalg'
        target_dir = PosixPath('/private/var/folders/24/q83vy1q54_181scf16svmls40000gn/T/pytest-of-tuhinsharma/pytest-10/test_cython0/linalg/tests/_cython_examples/build')
        tmp_path   = PosixPath('/private/var/folders/24/q83vy1q54_181scf16svmls40000gn/T/pytest-of-tuhinsharma/pytest-10/test_cython0')
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

popenargs = (['meson', 'compile', '-vv'],), kwargs = {'cwd': PosixPath('/private/var/folders/24/q83vy1q54_181scf16svmls40000gn/T/pytest-of-tuhinsharma/pytest-10/test_cython0/linalg/tests/_cython_examples/build')}
retcode = 1, cmd = ['meson', 'compile', '-vv']

    def check_call(*popenargs, **kwargs):
        """Run command with arguments.  Wait for command to complete.  If
        the exit code was zero then return, otherwise raise
        CalledProcessError.  The CalledProcessError object will have the
        return code in the returncode attribute.

        The arguments are the same as for the call function.  Example:

        check_call(["ls", "-l"])
        """
        retcode = call(*popenargs, **kwargs)
        if retcode:
            cmd = kwargs.get("args")
            if cmd is None:
                cmd = popenargs[0]
>           raise CalledProcessError(retcode, cmd)
E           subprocess.CalledProcessError: Command '['meson', 'compile', '-vv']' returned non-zero exit status 1.

cmd        = ['meson', 'compile', '-vv']
kwargs     = {'cwd': PosixPath('/private/var/folders/24/q83vy1q54_181scf16svmls40000gn/T/pytest-of-tuhinsharma/pytest-10/test_cython0/linalg/tests/_cython_examples/build')}
popenargs  = (['meson', 'compile', '-vv'],)
retcode    = 1

/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/subprocess.py:413: CalledProcessError
---------------------------------------------------------------------------------------------------- Captured stdout call ----------------------------------------------------------------------------------------------------
The Meson build system
Version: 1.4.0
Source dir: /private/var/folders/24/q83vy1q54_181scf16svmls40000gn/T/pytest-of-tuhinsharma/pytest-10/test_cython0/linalg/tests/_cython_examples
Build dir: /private/var/folders/24/q83vy1q54_181scf16svmls40000gn/T/pytest-of-tuhinsharma/pytest-10/test_cython0/linalg/tests/_cython_examples/build
Build type: native build
Project name: random-build-examples
Project version: undefined
C compiler for the host machine: cc (clang 15.0.0 "Apple clang version 15.0.0 (clang-1500.3.9.4)")
C linker for the host machine: cc ld64 1053.12
C++ compiler for the host machine: c++ (clang 15.0.0 "Apple clang version 15.0.0 (clang-1500.3.9.4)")
C++ linker for the host machine: c++ ld64 1053.12
Cython compiler for the host machine: cython (cython 3.0.10)
Host machine cpu family: x86_64
Host machine cpu: x86_64
Program python3 found: YES (/Users/tuhinsharma/.virtualenvs/scipy/bin/python)
Found pkg-config: YES (/usr/local/bin/pkg-config) 0.29.2
Run-time dependency python found: YES 3.11
Build targets in project: 3

Found ninja-1.11.1.git.kitware.jobserver-1 at /Users/tuhinsharma/.virtualenvs/scipy/bin/ninja
[1/7] /Users/tuhinsharma/.virtualenvs/scipy/bin/meson --internal copy ../extending.pyx extending_cpp.pyx
[2/7] cython -M --fast-fail -3 /private/var/folders/24/q83vy1q54_181scf16svmls40000gn/T/pytest-of-tuhinsharma/pytest-10/test_cython0/linalg/tests/_cython_examples/extending.pyx -o extending.cpython-311-darwin.so.p/extending.pyx.c
FAILED: extending.cpython-311-darwin.so.p/extending.pyx.c
cython -M --fast-fail -3 /private/var/folders/24/q83vy1q54_181scf16svmls40000gn/T/pytest-of-tuhinsharma/pytest-10/test_cython0/linalg/tests/_cython_examples/extending.pyx -o extending.cpython-311-darwin.so.p/extending.pyx.c

Error compiling Cython file:
------------------------------------------------------------
...
#!/usr/bin/env python3
#cython: language_level=3
#cython: boundscheck=False
#cython: wraparound=False

from scipy.linalg.cython_blas cimport cdotu
^
------------------------------------------------------------

/private/var/folders/24/q83vy1q54_181scf16svmls40000gn/T/pytest-of-tuhinsharma/pytest-10/test_cython0/linalg/tests/_cython_examples/extending.pyx:6:0: 'scipy/linalg/cython_blas.pxd' not found
[3/7] cython -M --fast-fail -3 --cplus extending_cpp.pyx -o extending_cpp.cpython-311-darwin.so.p/extending_cpp.pyx.cpp
FAILED: extending_cpp.cpython-311-darwin.so.p/extending_cpp.pyx.cpp
cython -M --fast-fail -3 --cplus extending_cpp.pyx -o extending_cpp.cpython-311-darwin.so.p/extending_cpp.pyx.cpp

Error compiling Cython file:
------------------------------------------------------------
...
#!/usr/bin/env python3
#cython: language_level=3
#cython: boundscheck=False
#cython: wraparound=False

from scipy.linalg.cython_blas cimport cdotu
^
------------------------------------------------------------

extending_cpp.pyx:6:0: 'scipy/linalg/cython_blas.pxd' not found
ninja: build stopped: subcommand failed.
INFO: autodetecting backend as ninja
INFO: calculating backend command to run: /Users/tuhinsharma/.virtualenvs/scipy/bin/ninja -v
================================================================================================== short test summary info ===================================================================================================
FAILED scipy/linalg/tests/test_extending.py::test_cython - subprocess.CalledProcessError: Command '['meson', 'compile', '-vv']' returned non-zero exit status 1.
===================================================================================================== 1 failed in 4.51s ======================================================================================================
```


### SciPy/NumPy/Python version and system information

```shell
Machine - Intel x86_64 MacOS Sonoma 14.4.1
Scipy - latest from the main branch
numpy==1.26.4
Python - 3.11.8
```

--- a/scipy/_lib/_testutils.py
+++ b/scipy/_lib/_testutils.py
@@ -13,6 +13,7 @@ import sysconfig
 from importlib.util import module_from_spec, spec_from_file_location
 
 import numpy as np
+import scipy
 
 try:
     # Need type: ignore[import-untyped] for mypy >= 1.6
@@ -41,6 +42,9 @@ IS_MUSL = False
 _v = sysconfig.get_config_var('HOST_GNU_TYPE') or ''
 if 'musl' in _v:
     IS_MUSL = True
+
+
+IS_EDITABLE = 'editable' in scipy.__path__[0]
 
 
 class FPUModeChangeWarning(RuntimeWarning):

--- a/scipy/linalg/tests/test_extending.py
+++ b/scipy/linalg/tests/test_extending.py
@@ -4,11 +4,13 @@ import platform
 import numpy as np
 import pytest
 
-from scipy._lib._testutils import _test_cython_extension, cython
+from scipy._lib._testutils import IS_EDITABLE, _test_cython_extension, cython
 from scipy.linalg.blas import cdotu  # type: ignore[attr-defined]
 from scipy.linalg.lapack import dgtsv  # type: ignore[attr-defined]
 
 
+@pytest.mark.skipif(IS_EDITABLE,
+                    reason='Editable install cannot find .pxd headers.')
 @pytest.mark.skipif(platform.machine() in ["wasm32", "wasm64"],
                     reason="Can't start subprocess")
 @pytest.mark.skipif(cython is None, reason="requires cython")

--- a/scipy/optimize/tests/test_extending.py
+++ b/scipy/optimize/tests/test_extending.py
@@ -3,9 +3,11 @@ import platform
 
 import pytest
 
-from scipy._lib._testutils import _test_cython_extension, cython
+from scipy._lib._testutils import IS_EDITABLE, _test_cython_extension, cython
 
 
+@pytest.mark.skipif(IS_EDITABLE,
+                    reason='Editable install cannot find .pxd headers.')
 @pytest.mark.skipif(platform.machine() in ["wasm32", "wasm64"],
                     reason="Can't start subprocess")
 @pytest.mark.skipif(cython is None, reason="requires cython")

--- a/scipy/special/tests/test_extending.py
+++ b/scipy/special/tests/test_extending.py
@@ -3,10 +3,12 @@ import platform
 
 import pytest
 
-from scipy._lib._testutils import _test_cython_extension, cython
+from scipy._lib._testutils import IS_EDITABLE,_test_cython_extension, cython
 from scipy.special import beta, gamma
 
 
+@pytest.mark.skipif(IS_EDITABLE,
+                    reason='Editable install cannot find .pxd headers.')
 @pytest.mark.skipif(platform.machine() in ["wasm32", "wasm64"],
                     reason="Can't start subprocess")
 @pytest.mark.skipif(cython is None, reason="requires cython")

